File: python/86.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def partition(self, head, x):
        """
        :type head: ListNode
        :type x: int
        :rtype: ListNode
        """
        dumSmall,dumGreat=ListNode(-1),ListNode(-1)
        smaller,greater= dumSmall,dumGreat
        while head!=None:
            if head.val<x:
                smaller.next=head
                smaller=smaller.next
                
            else:
                greater.next=head
                greater=greater.next
            head=head.next
        smaller.next=dumGreat.next
        greater.next=None
        return dumSmall.next
        
        
        

File: java/Linked_list/86.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode partition(ListNode head, int x) {
        ListNode l=new ListNode(0);
        ListNode s=new ListNode(0);
        ListNode p1=l;
        ListNode p2=s;
        while(head!=null){
            if (head.val<x){
                p2.next=head;
                p2=p2.next;
            }
            else{
                p1.next=head;
                p1=p1.next;
            }
            head=head.next;
        }
        p1.next=null;
        p2.next=l.next;
        return s.next;
    }
}